spec/unit/transaction/persistence_spec.rb

```diff
@@ -175,4 +175,46 @@ def write_state_file(contents)
       expect(persistence.get_system_value(resource, property)).to eq(value)
     end
   end
+
+  describe "when checking if persistence is enabled" do
+    let(:mock_catalog) do
+      mock
+    end
+
+    before :each do
+      @persistence = Puppet::Transaction::Persistence.new
+    end
```

Comment:
I noticed 2 things after I merged this. Could you change this to using a `let(:persistence)` block? That way rspec will memoize the variable once for each test and ensure that state never leaks across rspec tests.

Also it looks like the test is sensitive to whatever persistence file is present locally, so I was getting a test failure with the following:

```
$ cat ~/.puppetlabs/opt/puppet/cache/state/transactionstore.yaml
---
resources:
  File[/tmp/bar]:
    parameters:
      ensure:
        system_value:
        - :file
$ bundle exec rspec spec/unit/transaction_spec.rb:157
...
  1) Puppet::Transaction when evaluating a resource when evaluating a skipped resource for corrective change it should persist in the transactionstore
     Failure/Error: expect(persistence.get_system_value(resource.ref, "message")).to eq(["foobar"])
       expected: ["foobar"]
            got: nil
       (compared using ==)
     # ./spec/unit/transaction_spec.rb:157:in `block (4 levels) in <top (required)>'
     # util/rspec_runner:44:in `run'
     # util/rspec_runner:59:in `<main>'
Finished in 21.2 seconds (files took 4.34 seconds to load)
```

Can you mock what the persistence returns so that it's not reading YAML from disk?

Reply:
Thanks @joshcooper. I'll stage a new PR changing the before to a let block.